### PR TITLE
Mkantzer/ricotta 26 2 reinit react and docker

### PR DESCRIPTION
**Pull request by mkantzer**
Better re-do of https://github.com/Lasagna-Love-Portal/project-ricotta/pull/27

Per the conversation during code-and-coffee, this pr:
- removes the previous setup
- reinitializes a next.js app with:
  - typescript
  - tailwind
  - eslint
  - `src/` directory
- configures `next build` to output static content (will be used later to shove into an nginx image)
- creates a multistage dockerfile with optimized caching
- docker-compose file for building and running app in dev mode. Also includes file watch (experimental in latest docker desktop)
- stub for additional documentation (can be filled out here or as another PR)

A note: this pr _does_ remove all frontend work that has already been done. I am basically clueless on frontend, so I'd probably need someone else to actually port that work over.

### CODE REVIEW

File: .github/workflows/super-linter.yml
Comment (pid1) [inline .github/workflows/super-linter.yml:1]: We need to ensure this doesn't get wiped out.

File: DEVELOPMENT.md
Comment (pid1) [inline DEVELOPMENT.md:1]: Can we include some rudimentary instructions here? Just a first pass so that the frontend folks can get to work on the backport.

**Review by pid1 — COMMENTED** [review]
Overall, I think this looks fine, see minor feedback above.

As long as the frontend folks have given signoff on the JS stack, this looks good.

File: .github/workflows/super-linter.yml
Comment (mkantzer) [reply .github/workflows/super-linter.yml:1]: pulled it back in

File: DEVELOPMENT.md
Comment (mkantzer) [reply DEVELOPMENT.md:1]: I've just added some: let me know if they need any additional work

**Review by mattkuznicki-ll — CHANGES_REQUESTED** [review]
Hi @mkantzer thanks for posting this up! I see your note about needing to bring the existing front-end work back into this, and I'm also pretty clueless on how to do that. The Docker side looks generally well to me, with the acknowledgement that instructions would be very helpful for developers.
One thing that should be easy is to bring back in the reference_materials folder which looks to have been removed in this PR. The materials in there are going to be useful for implementers, and I believe there's general agreement we're going to use the project-ricotta repo as the "first place to look" for general Project Ricotta information.

**Comment by mattkuznicki-ll** [comment]
@allyson-s-code would you be able to help @mkantzer with bringing back in the frontend work that this PR pushes out through its React re-init? Would it be as easy as bringing the files and changes in https://github.com/Lasagna-Love-Portal/project-ricotta/pull/22 back in, do you know? Or would it be more complicated?

**Comment by allyson-s-code** [comment]
@mattkuznicki-ll I'm just reading through all of this now.  I'm pretty new to React apps and I've never used Next so I'm not entirely sure how hard it will be to bring in the files.

File: DEVELOPMENT.md
Comment (mattkuznicki-ll) [reply DEVELOPMENT.md:1]: 👍

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Development Information and Instructions
+
+## Getting Started
+
+### Development Dependencies  
+
+The primary tool used for building and running this repo is [Docker](https://www.docker.com).
+
+Some tasks may require the usage of a local [npm](https://www.npmjs.com/package/npm) installation, but we hope to remove that in the future.
+
```
Comment (mattkuznicki-ll) [inline DEVELOPMENT.md:10]: I had to do a little bit of work to get Docker running on a Windows 11 Home instance I was testing this out on. Here's a suggestion with what I did:

```suggestion

For Windows users:
You can install Docker Desktop from https://docs.docker.com/desktop/install/windows-install/.
When running on Windows, you'll need to ensure the Windows Subsystem for Linux (WSL) is installed and updated to WSL version 2. To do this:
1. Open a Powershell windows as administrator
2. Install WSL and a Linux distribution. To install Ubuntu 22.04:
    `wsl --install -d Ubuntu-22.04`
3. To make sure WSL is updated to WSL 2:
    `wsl --update`
4. Then make sure Docker Desktop sees the installed WSL and Linux distribution. In Docker Desktop's Settings | Resources | WSL integration, make sure the Ubuntu 22.04 distribution is enabled. You may need to hit Refresh and wait a minute or two for it to show. Once enabled, hit Apply & Restart to restart the Docker engine.

```

**Comment by mattkuznicki-ll** [comment]
> @mattkuznicki-ll I'm just reading through all of this now. I'm pretty new to React apps and I've never used Next so I'm not entirely sure how hard it will be to bring in the files. I'm willing to give it a shot!

@allyson-s-code I'm pretty new to them also. If you're willing to give it a shot then Thank You! 🥇 and I think that can be a separate pull request from this one.

**Comment by mkantzer** [comment]
Updated:
- development docs to streamline and clarify workflows
- restored reference_materials

**Pull request merged**